main.go

```diff
@@ -435,25 +435,29 @@ func fetchManifestAndGenerateSDK(c *ContractConfig, host string) error {
 		return err
 	}
 
-	languages := cfg.Defaults.OnChain.Languages
-	if c.OnChain != nil {
-		languages = c.OnChain.Languages
-	}
-	for _, l := range languages {
-		err = generateSDK(&generators.GenerateCfg{Manifest: m, ContractHash: c.ScriptHash, SdkDestination: cfg.getSdkDestination(l)}, l, SDK_ONCHAIN)
-		if err != nil {
-			return err
+	if cfg.Defaults.OnChain != nil {
```

Comment:
The config defaults are optional. This checks prevents using a `cpm.yaml` file where `on-chain` is missing from the defaults section, but where the `contract` section does specify an `on-chain` section. e.g.
```
defaults:
  contract-source-network: mainnet
  contract-destination: neo-express
  contract-generate-sdk: true
contracts:
  - label: otter
    script-hash: '0x963e74226c0cffa8a2369116f298579f9721b09d'
    on-chain:
      languages:
        - python
...
```

Without this check it would have failed on the next line (which was probably the bug you encountered). I think we need to do something like the following
```go
	var languages []string
	if c.OffChain != nil {
		languages = c.OffChain.Languages
	} else if cfg.Defaults.OffChain != nil {
		languages = cfg.Defaults.OffChain.Languages
	} else {
		return fmt.Errorf("no off-chain SDK language specified. Add the 'off-chain' key to either the 'defaults'" +
			" or 'contract' specific section of your 'cpm.yaml' file")
	}
	for _, l := range languages {
		err = generateSDK(...)
```